src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -356,9 +361,29 @@ public SegmentMetadata.ComponentMetadataMap writeData(IndexDescriptor indexDescr
                                                         ? x -> ordinalMap.inverse().getOrDefault(x, x)
                                                         : x -> x;
 
+            // Retrieve the first compressed vectors with more than MAX_PQ_TRAINING_SET_SIZE rows
+            var indexes = new ArrayList<>(indexContext.getView().getIndexes());
+            indexes.sort(Comparator.comparing(ssTableIndex -> ssTableIndex.getSSTable().descriptor.id));
+            CompressedVectors compressedVectors = null;
+            outer:
+            for (SSTableIndex index : indexes)
+            {
+                for (Segment segment : index.getSegments())
+                {
+                    if (segment.metadata.numRows < ProductQuantization.MAX_PQ_TRAINING_SET_SIZE)
+                        continue;
+
+                    var searcher = segment.getIndexSearcher();
+                    assert searcher instanceof CompressedVectorsSupplier;
+                    compressedVectors = ((CompressedVectorsSupplier) searcher).getCompressedVectors();
+                    // We take the first compressed vectors with enough vectors and stop
+                    break outer;
+                }
+            }
+
```

Comment:
Need to add the pass for "if cv == null set it to the one from the largest segment" ?

Reply:
Thanks, I forgot about this case.